### PR TITLE
refactor: make CI green again!

### PR DESCRIPTION
**Pull request by MilkeeyCat**
Ive made some easy fixed, the rest is for you :)

### CODE REVIEW

File: src/rich_text.ts [hunk 1]
```diff
@@ -105,9 +105,11 @@ export const enrichText = (userinput: string): string => {
     }
   })
   userinput = userinput.replaceAll(
-    new RegExp(/(`{1,3})(.*?)(\1)|(https?:\/\/[a-zA-Z0-9\-_\[\]\?\#\:\&\$\+\*\%\/\.\=\@]+)/ig),
-    (match, _openingBackticks, _closingBackticks, _textInBackticks, url) => {
-      if (url == undefined) { return match }
+    // new RegExp(/(`{1,3})(.*?)(\1)|(https?:\/\/[a-zA-Z0-9-_[\]?#:&$+*%.=@]+)/ig),
+    /(`{1,3})(.*?)(\1)|(https?:\/\/[a-zA-Z0-9\-_[\]?#:&$+*%/.=@]+)/ig,
```
Comment (ChillerDragon) [inline src/rich_text.ts:109]: did you test if this still works?

File: src/autocomplete.ts [hunk 2]
```diff
@@ -42,9 +42,9 @@ export const autoComplete = (
   event: KeyboardEvent,
   inputField: HTMLInputElement,
   suffix: string = ''
-) => {
+): void => {
   const end: number = inputField.value.length
-  const compState: CompletionState = completionStates[prefix] || {
+  const compState: CompletionState = completionStates[prefix] ?? {
```
[the remark this answers is not in the document]
Comment (MilkeeyCat) [reply src/autocomplete.ts:47]: `??` means if the thing on the left exists then take it otherswise take whats on the right side

File: src/rich_text.ts (hunk 1 above)
Comment (MilkeeyCat) [reply src/rich_text.ts:109]: no but i hope it will. you can run the project because of the errors :D

Comment (ChillerDragon) [reply src/rich_text.ts:109]: Oh true. Lets get rid of the tsconfig update until it is fully fixed. Failing style pipeline is bad enough. But failing build is not okay. I remember something about a different amount of escapes or backslashes in // vs as RegExp string. Might not be an issue here but would feel safer if this was tested.

File: src/autocomplete.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChillerDragon) [reply src/autocomplete.ts:47]: Okay thats what I kinda assumed. Thanks for being my google :heart: 

So its the same as || but it doesnt crash

``completionStates[prefix]``
even if ``completionStates`` is ``null``?

File: src/rich_text.ts (hunk 1 above)
Comment (MilkeeyCat) [reply src/rich_text.ts:109]: i tested it, it seems like everything works as expected

Comment (ChillerDragon) [reply src/rich_text.ts:108]: poggies

**Comment by ChillerDragon** [comment]
Thanks

**Pull request merged**